sirepo/sim_data/flash.py

```diff
@@ -275,10 +275,12 @@ def __put_sim_files(cls, data, datafiles, make_dir, run_dir):
             .items()
         ):
             p = make_dir.join(c)
-            cls.delete_sim_file(
-                data.models.simulation.simulationId,
-                b.split(cls._FLASH_FILE_NAME_SEP)[0],
-            )
+            # TODO(pjm): this call broke the LaserSlab example
```

Comment:
@moellep did you see this in dev? I can't recreate it. Wondering if it is an NFS issue on prod.

Reply:
Yes, I was getting this repeatedly on dev. I originally thought the files weren't making it through the sim_db "put", but found that removing the delete and they arrived ok. I think somehow the delete and put calls were being processed out of order, so the delete was happening after the put. I don't know why the delete would be necessary at all as the put would be overwriting the files.

Reply:
Delete is necessary to clean up the flash exe's. The exe's are named like  flash_exe_uniqueHash. The put file might not have the same uniqueHash (probably won't).  But, looking at the code I think we no longer need to have the hash. It looks like you changed the gui so if any fields are updated that require flash to be compiled again the user is forced to run the setupAnimation. Is that correct? If so, we can get rid of the hash and get rid of the delete.

Reply:
I'm not sure if you refresh the browser if it still knows it has been modified, so that may be necessary. We could change the code so it only calls the delete on the flash exe (only if the hash has changed)?

Reply:
This is actually a bug somewhere in delete sim_db_file. It looks like only a partial path is being received which causes multiple files to be deleted at once:
```
DELETE flash/KNeZLS2E/al
PUT flash/KNeZLS2E/al-imx-003.cn4
DELETE flash/KNeZLS2E/he
PUT flash/KNeZLS2E/he-imx-005.cn4
DELETE flash/KNeZLS2E/al
PUT flash/KNeZLS2E/al-imx-003.ses
DELETE flash/KNeZLS2E/he
PUT flash/KNeZLS2E/he-imx-005.ses
DELETE flash/KNeZLS2E/amr_runtime_parameters.tpl
PUT flash/KNeZLS2E/amr_runtime_parameters.tpl
DELETE flash/KNeZLS2E/flash_exe
PUT flash/KNeZLS2E/flash_exe-327872c1931944b92c097bedd3218e7a
```
Notice the file names with dashes are being truncated which causes anything that matches it to get deleted. This bug causes Laser Slab to fail when you try to run the simulation becuase some of the *.cn4 are missing.

Reply:
I've updated the code so it only deletes the flash executable file. Otherwise files with "-" character would be processed incorrectly.

Reply:
Aha. We truncate the filenames when we [send the delete request](https://github.com/radiasoft/sirepo/blob/359f584282c32ea3df7c3cbffffea687d0225d1d/sirepo/sim_data/flash.py#L280). And then do a glob [in the delete in the supervisor](https://github.com/radiasoft/sirepo/blob/359f584282c32ea3df7c3cbffffea687d0225d1d/sirepo/sim_db_file.py#L25). So, that behavior needs to be rethought.

Looking at the log you share we put `al-imx-003.cn4` and then call delete on `al*` before doing the put for `al-imx-003.ses`. So, `al-imx-003.cn4` is deleted even though we just put it and need it to run the sim. 

I need to rethink the code. At first glance I think we should only do the glob on the flash exe. It would be nice to not glob at all but we don't have a way of re-creating the hash for the flash_exe so I'm not sure how we'll handle that. Maybe we can hash the whole binary and send that info back and forth to decide if we need to do a recreate/delete.

Another idea is to do the deletes and then do the puts after. That way they won't conflict. That is probably the simplest option.

Reply:
Sorry GH didn't update and I didn't see your message before sending mine. You solution sounds good.